### PR TITLE
Fix race condition on GPU

### PR DESCRIPTION
**Pull request by ymwangg**
This is a temporary fix for https://github.com/pytorch/xla/issues/4541 until it can be fixed on tensorflow side.

Btw: this PR should also fix the flaky ddp test on GPU.

### CODE REVIEW

File: tf_patches/gpu_race_condition.diff [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+index ed55b32c47a..260fddb3f5c 100644
+--- a/tensorflow/compiler/xla/service/gpu/gpu_executable.cc
++++ b/tensorflow/compiler/xla/service/gpu/gpu_executable.cc
+@@ -489,8 +489,7 @@ StatusOr<ExecutionOutput> GpuExecutable::ExecuteAsyncOnStreamImpl(
+       "GpuExecutable::ExecuteAsyncOnStreamImpl(", module_name_, ")"));
+   se::DeviceMemoryAllocator* const memory_allocator = run_options->allocator();
+   // Force synchronous execution if the allocator requires it.
+-  const bool block_host_until_done =
+-      !memory_allocator->AllowsAsynchronousDeallocation();
++  const bool block_host_until_done = true;
```
Comment (JackCaoG) [inline tf_patches/gpu_race_condition.diff:11]: Will this has speed implication?

Comment (ymwangg) [reply tf_patches/gpu_race_condition.diff:11]: Appears not. For all the models I have tested: resnet50, ViT, bert, the speed remains the same with this patch.

**Review by JackCaoG — APPROVED** [review]

**Comment by JackCaoG** [comment]
Thanks @ymwangg !

**Comment by JackCaoG** [comment]
FYI @alanwaketan since we can enable the DDP test for GPU now.

**Pull request merged**